src/components/buttons/FilledButton.tsx

```diff
@@ -1,32 +1,64 @@
 import React from 'react';
-import type { TextStyle, ViewStyle } from 'react-native';
+import { StyleSheet, ViewStyle } from 'react-native';
 import { Button as BaseButton } from 'react-native-paper';
 import { useTheme } from '../../styles/themes';
 import type { BaseButtonProps } from './BaseButtonProps';
 
-export const FilledButton = ({ children, style, labelStyle, onClick, testID }: BaseButtonProps) => {
+export const FilledButton = ({
+    children,
+    style,
+    labelStyle,
+    onClick,
+    testID,
+    disabled,
+    status,
+}: BaseButtonProps) => {
     const theme = useTheme();
-    const buttonStyle: ViewStyle = {
-        borderRadius: 8,
-        backgroundColor: theme.sw.colors.primary[400],
-        ...style,
-    };
 
-    const _labelStyle: TextStyle = {
-        fontFamily: 'PublicSans-Regular',
-        fontSize: 16,
-        lineHeight: 19,
-        color: theme.sw.colors.neutral[50],
-        paddingVertical: theme.sw.spacing.s,
-        paddingHorizontal: theme.sw.spacing.l,
-        // Overrides default margin of Paper component
-        marginVertical: 0,
-        marginHorizontal: 0,
-        ...labelStyle,
+    const getbackgroundColor = (): ViewStyle['backgroundColor'] => {
+        switch (status) {
```

Comment:
Could you move this function in it's own file (like `utils`), so that we can reuse it in the 3 button variants ?

Something like `getColor(status) -> string` so it could be reused without being concerned wether we want to set the background color, the text color or the border color ?

Reply:
updated :)